src/main/resources/templates/recipes.html

```diff
@@ -6,6 +6,7 @@
 </head>
 <body>
 <h3 th:text="${userName}"></h3>
-<h1>Super Secret Recipes</h1>
+<h1>Secret Recipes</h1>
+<ul th:each="${siteUser}"></ul>
```

Comment:
There are a couple things that will help here:

1. What do you actually want to repeat? Right now, you're creating a new unordered list element for each user. Instead, you'll want to create **one** unordered list, and then create list item elements for each user.
2. These Thymeleaf "each" loops function a lot like [for-each loops](https://www.programiz.com/java-programming/enhanced-for-loop) in Java. You're basically asking Thymeleaf to loop through an array of user objects, then access each object by assigning it to a variable. For example, `th:each="s : ${siteUser}"` goes through the list of `siteUser`s you passed in to this template and temporarily assigns each one to the variable `s`. It can then access that user's properties: `s.userName`, `s.recipes`, etc.
3. Use the `th:text` attribute to actually write those properties to the content of your list items.